subprojects/plugin-use/src/main/java/org/gradle/plugin/use/internal/DefaultPluginRequestApplicator.java

```diff
@@ -79,12 +80,15 @@ public DefaultPluginRequestApplicator(PluginRegistry pluginRegistry, PluginResol
         this.cachedClasspathTransformer = cachedClasspathTransformer;
     }
 
-    public void applyPlugins(final PluginRequests requests, final ScriptHandlerInternal scriptHandler, @Nullable final PluginManagerInternal target, final ClassLoaderScope classLoaderScope) {
-        if (requests.isEmpty()) {
+    public void applyPlugins(final PluginRequests requests, final ScriptHandlerInternal scriptHandler, ScriptTarget target, final ClassLoaderScope classLoaderScope) {
```

Comment:
:x: Requiring a `ScriptTarget` here just to get a flag out of it doesn't sound worth it. Consider that `kotlin-dsl` doesn't use `ScriptTarget` at all and all the baggage in that interface.

Reply:
How about extracting a PluginTarget?

Reply:
Sounds good 👍

Reply:
An alternative design would be to change the protocol a bit and expose the `defineScriptHandlerClassScope(scriptHandler, classLoaderScope, Collections.<PluginImplementation<?>>emptyList());` code path as a public operation so that clients that already know the target does not support the plugins block could call it directly.